### ISSUE TITLE
RMSNorm Implementation Question

### ISSUE THREAD

**Issue by OAHC2022** [lucidrains/denoising-diffusion-pytorch#333]
Here is the code in the [denoising_diffusion_pytorch_1d.py](https://github.com/lucidrains/denoising-diffusion-pytorch/blob/ec0a1c7596f654d9b5d3952a63ce3301128f1979/denoising_diffusion_pytorch/denoising_diffusion_pytorch_1d.py#L110)
```python
class RMSNorm(Module):
    def __init__(self, dim):
        super().__init__()
        self.g = nn.Parameter(torch.ones(1, dim, 1))

    def forward(self, x):
        return F.normalize(x, dim = 1) * self.g * (x.shape[1] ** 0.5)
```
I am wondering if it should instead be `x.shape[1] ** -0.5` given the equation is 
![image](https://github.com/user-attachments/assets/80161a14-daa4-4e67-8f39-9792cbc92c22)

Also, I am curious why this normalization is applied at dim=1. If the shape of x is three-dimensional (same as `self.g`), what about the last dimension?
Thanks!

**Comment by OAHC2022** [comment]
I realized that I overlooked the entire equation where `sqrt(1/n)` is in the denominator:
![image](https://github.com/user-attachments/assets/aff16870-6093-442f-a5c7-c89f26d87db7)
Sorry about this. I will close this issue.
